### PR TITLE
[gui] Tolerate swapchain image count

### PR DESCRIPTION
**Pull request by PENGUINLIONG**
Issue: #6581

### CODE REVIEW

File: taichi/rhi/vulkan/vulkan_device.cpp [hunk 1]
```diff
@@ -2438,7 +2438,7 @@ void VulkanSurface::create_swap_chain() {
   createInfo.pNext = nullptr;
   createInfo.flags = 0;
   createInfo.surface = surface_;
-  createInfo.minImageCount = capabilities.minImageCount;
+  createInfo.minImageCount = std::min<uint32_t>(capabilities.maxImageCount, 3);
```
Comment (ailzhang) [inline taichi/rhi/vulkan/vulkan_device.cpp:2441]: nit: would you mind adding a comment explaining why `3` is chosen :P? Thanks!

**Review by ailzhang — APPROVED** [review]
Thanks!

File: taichi/rhi/vulkan/vulkan_device.cpp (hunk 1 above)
Comment (PENGUINLIONG) [reply taichi/rhi/vulkan/vulkan_device.cpp:2441]: @ailzhang https://www.intel.com/content/www/us/en/support/articles/000006930/graphics.html

Comment (PENGUINLIONG) [reply taichi/rhi/vulkan/vulkan_device.cpp:2441]: it's a rather common practice actually, you know it if you are a gamer 😎

**Comment by amakiir** [comment]
Can I ask why the commit was reverted? Because the commit fixed the `Assertion failed: info->MinImageCount >= 2` bug for me. This means the bug is not fixed in the current (1.3.0) release.

**Pull request merged**